src/Symfony/Component/ExpressionLanguage/ParserCache/PSR6ParserCacheAdapter.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\ParserCache;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheItemInterface;
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * @author Alexandre GESLIN <alexandre@gesl.in>
+ */
+class PSR6ParserCacheAdapter implements CacheItemPoolInterface
+{
+    /**
+     * @var ParserCacheInterface
+     */
+    private $pool;
+
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    public function __construct(ParserCacheInterface $pool)
+    {
+        $this->pool = $pool;
+        $defaultLifetime = 0;
+
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) use ($defaultLifetime) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+                $item->defaultLifetime = $defaultLifetime;
+
+                return $item;
+            },
+            null,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $isHit = false;
+
+        if (null !== $value = $this->pool->fetch($key)) {
+            $isHit = true;
+        }
+
+        $f = $this->createCacheItem;
+
+        return $f($key, $value, $isHit);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        $this->pool->save($item->getKey(), $item->get());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = array())
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+    }
```

Comment:
I was wondering, if this adapter is used internally only, to adapt CacheItemPoolInterface only, what's the aim to make unnecessary code ? But it is part of my interrogations